vllm/worker/vineyard_llm_cache.py

```diff
@@ -193,9 +215,19 @@ def prefetch_seq_kv_caches(
             # torch.distributed.broadcast(slot_mapping, src=0,
             #                             group=get_tensor_model_parallel_group())
 
-
+        self.metrics.hit_tokens += matched
+        self.metrics.total_tokens += len(tokens)
+        self.metrics.hit_blocks += (matched // block_size)
+        self.metrics.total_blocks += (- len(tokens) // (-block_size))
+        self.metrics.counter += 1
         # save to GPU kv cache
+        torch.cuda.synchronize()
+        start_time = time.time()
         buffer = self.buffer[:, :, offset:offset+matched].cuda()
+        torch.cuda.synchronize()
+        duration = time.time() - start_time
+        self.metrics.time_load.append(duration)
+        self.metrics.time_load.append(0 if matched == 0 else duration/matched)
```

Comment:
Why do you append two measurements to `time_load`? If you want to normalized the loading time by the number of tokens matched, then you should probably used a separate metrics field?